docs/src/dplyr.md

```diff
@@ -0,0 +1,355 @@
+
+# Data: Mammals Sleep
+
+The `msleep` (mammals sleep) data set contains the sleep times and weights for a set of mammals and is available in the dagdata repository on GitHub. This data set contains 83 rows and 11 variables.  
+
+We can load the data directly into a DataFrame from the `url`. 
+
+```@repl 1
+url = "https://raw.githubusercontent.com/genomicsclass/dagdata/master/inst/extdata/msleep_ggplot2.csv"
+msleep = CSV.read(HTTP.get(url).body, DataFrame; missingstring="NA")
+```
+
+The columns (in order) correspond to the following: 
+
+column name | Description
+--- | ---
+`:name` | common name
+`:genus` | taxonomic rank
+`:vore` | carnivore, omnivore or herbivore?
+`:order` | taxonomic rank
+`:conservation` | the conservation status of the mammal
+`:sleep_total` | total amount of sleep, in hours
+`:sleep_rem` | rem sleep, in hours
+`:sleep_cycle` | length of sleep cycle, in hours
+`:awake` | amount of time spent awake, in hours
+`:brainwt` | brain weight in kilograms
+`:bodywt` | body weight in kilograms
+
+
+# Important DataFramesMeta.jl Verbs To Remember
+
+Many DataFrames.jl macros come in two forms, a version which operates on columns as a whole and a version which operations row-wise, prefixed by `r`.
+
+DataFramesMeta.jl macro | By-row version | Description | `dplyr` equivalent
+--- | --- | --- | ---
+`@select` | `@rselect`| select columns | `select`
+`@transform` | `@rtransform` | create new columns | `mutate`
+`@subset` | `@rsubset` | filter rows | `filter`
+`@orderby` | `@rorderby` | re-order or arrange rows | `arrange`
+`@combine` | | summarise values | `summarize`
+`groupby` | | allows for group operations in the "split-apply-combine" concept | `group_by`
+
+# DataFramesMeta.jl Verbs In Action
+
+Two of the most basic functions are `@select` and `@subset`, which selects columns and filters rows respectively. To reference columns, use the `Symbol` of the column name, i.e. `:name` refers to the column `msleep.name`.
+
+## Selecting Columns Using `@select`
+
+Select a set of columns: the `:name` and the `:sleep_total` columns. 
+
+```@repl 1
+@select msleep :name :sleep_total
+```
+
+If you have a column name stored as a variable, you can select it as a column with the syntax `$`. 
+
+```@repl 1
+varname = :sleep_total
+@select msleep :name $varname
+```
+
+The `$` sign has special meaning in DataFramesMeta.jl. We use it for any column reference which is *not* a symbol. Without it, DataFramesMeta.jl can't tell whether `varname` refers to the column `:sleep_total`. 
+
+You can also use `$` to refer to columns with strings
+
+```@repl 1
+varname = "sleep_total"
+@select msleep :name $varname
+```
+
+as well as vectors of variable names 
+
+```@repl 1
+varnames = ["name", "sleep_total"]
+@select msleep $varnames
+```
+
+Similarly, to select the first column, use the syntax `$1`. 
```

Comment:
yes.

Right now, the parsing for selecting columns is exactly the same as working with anonymous functions. So since `@transform df :y = :x .+ 1` would be ambiguous if we allowed `1` to be a column selector in the anonymous function, we need the same thing when doing `select`. 

Not ideal, though. We can change this before 1.0.

Reply:
I guess it is :).

Reply:
I think `$1` makes sense - it just probably should be well explained somewhere.